async-nats/src/jetstream/kv/mod.rs

```diff
@@ -1180,6 +1234,59 @@ impl Display for StatusErrorKind {
 
 pub type StatusError = Error<StatusErrorKind>;
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum CreateErrorKind {
+    AlreadyExists,
+    InvalidKey,
+    Publish,
+    Ack,
+    Other,
+}
+
+impl From<UpdateError> for CreateError {
+    fn from(error: UpdateError) -> Self {
+        match error.kind() {
+            UpdateErrorKind::InvalidKey => Error::from(CreateErrorKind::InvalidKey),
+            UpdateErrorKind::TimedOut => Error::from(CreateErrorKind::Publish),
+            UpdateErrorKind::Other => Error::from(CreateErrorKind::Other),
+        }
+    }
+}
+
+impl From<PutError> for CreateError {
+    fn from(error: PutError) -> Self {
+        match error.kind() {
+            PutErrorKind::InvalidKey => Error::from(CreateErrorKind::InvalidKey),
+            PutErrorKind::Publish => Error::from(CreateErrorKind::Publish),
+            PutErrorKind::Ack => Error::from(CreateErrorKind::Ack),
+        }
+    }
+}
+
+impl From<EntryError> for CreateError {
+    fn from(error: EntryError) -> Self {
+        match error.kind() {
+            EntryErrorKind::InvalidKey => Error::from(CreateErrorKind::InvalidKey),
+            EntryErrorKind::TimedOut => Error::from(CreateErrorKind::Publish),
+            EntryErrorKind::Other => Error::from(CreateErrorKind::Other),
+        }
+    }
+}
+
+impl Display for CreateErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::AlreadyExists => write!(f, "key already exists"),
+            Self::Publish => write!(f, "failed to put key into store"),
+            Self::Ack => write!(f, "ack error"),
+            Self::InvalidKey => write!(f, "key cannot be empty or start/end with `.`"),
+            Self::Other => write!(f, "other error"),
```

Comment:
I would call it "failed creating a key"  to be consistent with other errors.

Reply:
Makes sense, made the change